pkgs/top-level/python-aliases.nix

```diff
@@ -231,6 +231,7 @@ mapAliases ({
   htmllaundry = throw "htmllaundry has been removed because it is abandoned"; # added 2024-06-04
   HTSeq = htseq; # added 2023-02-19
   hyperkitty = throw "Please use pkgs.mailmanPackages.hyperkitty"; # added 2022-04-29
+  icoextract = throw "icoextract has been promoted to a top-level attribute name: `pkgs.icoextract`"; # added 2024-07-07
```

Comment:
Could this package be used as a python module?

Reply:
[Could yes](https://github.com/bottlesdevs/Bottles/blob/eb6399eb55cbf57a53291e7d9c21042cea10d48b/bottles/backend/utils/manager.py#L179), but it's mainly used for its thumbnailer and CLI.

Motivation of this change is to make it [easier for users to use](https://github.com/quuinnn/NixOS-Configs/blob/bd2d052452fcca373cc515e6082e5332e691bae7/etc/nixos/configuration.nix#L183) since they don't have to remember that `icoextract` is inside `python311Packages`.

This PR takes inspiration from prior work such as [pdfposter: move out of python-modules](https://github.com/NixOS/nixpkgs/pull/240414/commits/39e70ab2a60aade7b6f638de9565574c3f0a75a2).

Reply:
Then don't write this, write

```nix
icoextract = toPythonModule (pkgs.icoextract.override {
  python3Packages = self;
});
```

in `top-level/python-packages.nix`